src/Uno.UI/UI/Xaml/Application.skia.cs

```diff
@@ -47,22 +51,91 @@ partial void InitializePartial()
 				throw new InvalidOperationException("The application must be started using Application.Start first, e.g. Microsoft.UI.Xaml.Application.Start(_ => new App());");
 			}
 
-			CoreApplication.SetInvalidateRender(compositionTarget =>
+			CoreApplication.SetInvalidateRender(OnInvalidateRender, OnSetContinuousRender);
+		}
+
+		private void OnSetContinuousRender(object? compositionTarget, bool enabled)
+		{
+			Debug.Assert(compositionTarget is null or CompositionTarget);
+
+			if (compositionTarget is CompositionTarget { Root: { } root })
 			{
-				Debug.Assert(compositionTarget is null or CompositionTarget);
+				var originalCount = _continuousTargets.Count;
 
-				if (compositionTarget is CompositionTarget { Root: { } root })
+				if (_continuousTargets.Contains(root))
 				{
-					foreach (var cRoot in CoreServices.Instance.ContentRootCoordinator.ContentRoots)
+					if (!enabled)
 					{
-						if (cRoot?.XamlRoot is { } xRoot && ReferenceEquals(xRoot.VisualTree.RootElement.Visual, root))
+						_continuousTargets.Remove(root);
+					}
+				}
+				else
+				{
+					if (enabled)
+					{
+						_continuousTargets.Add(root);
+					}
+				}
+
+				if (originalCount != _continuousTargets.Count)
+				{
+					if (_continuousTargets.Count == 0)
+					{
+						if (typeof(ApplicationLanguages).Log().IsEnabled(LogLevel.Trace))
+						{
+							typeof(ApplicationLanguages).Log().Trace($"OnSetContinuousRender({enabled}) detach (Count:{_continuousTargets.Count})");
```

Comment:
Why `typeof(ApplicationLanguages)` ?

```suggestion
						if (this.Log().IsEnabled(LogLevel.Trace))
						{
							this.Log().Trace($"OnSetContinuousRender({enabled}) detach (Count:{_continuousTargets.Count})");
```